### PR TITLE
Add type annotations in jina/clients

### PR DESCRIPTION
**Pull request by rameshwara**

### CODE REVIEW

**Comment by JoanFM** [comment]
Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
#553 :bookmark: Commit Message Guideline for the First Time Contributor
jina-ai/jina | Jun 21st | Added by GitHub
CONTRIBUTING.md
jina-ai/jina | Added by GitHub

**Comment by rameshwara** [comment]
> Bad commit message, please refer to:
> 
> * [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
> * [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
>   #553 🔖 Commit Message Guideline for the First Time Contributor
>   jina-ai/jina | Jun 21st | Added by GitHub
>   CONTRIBUTING.md
>   jina-ai/jina | Added by GitHub

Done, but the CI tests are still failing.

**Comment by JoanFM** [comment]
> > Bad commit message, please refer to:
> > 
> > * [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
> > * [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
> >   #553 bookmark Commit Message Guideline for the First Time Contributor
> >   jina-ai/jina | Jun 21st | Added by GitHub
> >   CONTRIBUTING.md
> >   jina-ai/jina | Added by GitHub
> 
> Done, but the CI tests are still failing.

Now, Lint complains that `PyClient` is not found,

To overcome this issue but still avoid this unnecesessary import, we use the trick to do

```python
if False:
    import whatever_missing_package_needed_for_type_annotation

```

File: jina/clients/__init__.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 __copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
+if False:
+    import PyClient
```
Comment (JoanFM) [inline jina/clients/__init__.py:5]: PyClient does not exist as package, the right one is `from .python import PyClient`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/clients/__init__.py (hunk 1 above)
Comment (rameshwara) [reply jina/clients/__init__.py:5]: Great. It's an interesting way of handling that issue.

Comment (rameshwara) [reply jina/clients/__init__.py:5]: > PyClient does not exist as package, the right one is `from .python import PyClient`

Alright will update that in the next commit.

**Comment by JoanFM** [comment]
Please keep one PR per package

**Comment by rameshwara** [comment]
> Please keep one PR per package

Alright, will do that from the next one. Sorry.

File: jina/docker/database.py [hunk 2]
```diff
@@ -49,21 +50,21 @@ def database(self):
     def collection(self):
         return self.database[self.collection_name]
     
-    def find(self, query: dict):
+    def find(self, query: Dict) -> None:
```
Comment (JoanFM) [inline jina/docker/database.py:53]: Is there a way to know the key and value types of this Dictionary?

Comment (rameshwara) [reply jina/docker/database.py:53]: Yes, its Dict[str, Union[Dict, List]]. Have made the addition.

File: jina/docker/helper.py [hunk 3]
```diff
@@ -36,12 +37,12 @@ def get_default_login():
     return login_info
 
 
-def handle_dot_in_keys(document):
+def handle_dot_in_keys(document: Dict[str, Union[Dict, List]]) -> Union[Dict, List]:
     updated_document = {}
     for key, value in document.items():
-        if isinstance(value, dict):
+        if isinstance(value, Dict):
```
Comment (JoanFM) [inline jina/docker/helper.py:43]: let's keep it as it was, type hinting is not enforced by Python, and I am not sure how interchangable in terms of functionality `Dict` vs `dict` is. These PRs should only affect function signatures' type hinting but not touch the function.

File: jina/docker/helper.py [hunk 4]
```diff
@@ -36,12 +37,12 @@ def get_default_login():
...
             value = handle_dot_in_keys(value)
-        if isinstance(value, list) and len(value) > 0 and isinstance(value[0], dict):
+        if isinstance(value, List) and len(value) > 0 and isinstance(value[0], dict):
```
Comment (JoanFM) [inline jina/docker/helper.py:45]: same here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/docker/helper.py (hunk 3 above)
Comment (hanxiao) [reply jina/docker/helper.py:43]: this is wrong, use `dict`

File: jina/docker/helper.py (hunk 4 above)
Comment (hanxiao) [reply jina/docker/helper.py:45]: this is wrong, use `list`

File: jina/docker/helper.py (hunk 3 above)
Comment (rameshwara) [reply jina/docker/helper.py]: Done

File: jina/docker/helper.py (hunk 4 above)
Comment (rameshwara) [reply jina/docker/helper.py]: Done

**Review by JoanFM — APPROVED** [review]

**Pull request merged**